rust/src/pgsql/pgsql.rs

```diff
@@ -505,10 +538,22 @@ impl PgsqlState {
                         needed_estimation,
                         &start
                     );
+                    self.set_event(PgsqlEvent::TruncatedData);
+                    SCLogDebug!("PgsqlEvent::TruncatedData");
                     return AppLayerResult::incomplete(consumed as u32, needed_estimation as u32);
                 }
+                Err(Err::Error(Error{code:ErrorKind::Verify, ..})) => {
+                    SCLogDebug!("PgsqlEvent::InvalidLength");
+                    self.set_event(PgsqlEvent::InvalidLength);
+                    AppLayerResult::err();
+                }
+                Err(Err::Error(Error{code:ErrorKind::Switch, ..})) => {
+                    self.set_event(PgsqlEvent::MalformedData);
+                    SCLogDebug!("PgsqlEvent::MalformedData");
+                    return AppLayerResult::ok();
```

Comment:
Must understand if I'm not seeing events with this one due to `UnknownMessage` type. But we only have that for BackEnd messages, so... must investigate. Personal note: tlpwhite jdbc pcap may help me here.